### ISSUE TITLE
Fix constant folding pass hang in eager mode

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#64]
## Problem

The StableHLO aggressive folder pass (`createStablehloAggressiveFolderPass`) causes execution hangs with certain patterns, specifically cholesky gradient in eager mode.

## Root Cause

The folder pass folds `broadcast_in_dim` of scalar constants into pre-broadcast constants (e.g., `constant<5> : tensor<ui32>` + `broadcast_in_dim` → `constant<5> : tensor<2xui32>`). This produces valid StableHLO IR, but causes infinite execution in MPS Graph's while-loop handling.

The while-body shape mismatch fix (reshaping body outputs to match initial input shapes, added in c4ad1c2) solves the crash/abort (`tensor<1xf32>` vs `tensor<f32>` mismatch), but the hang persists.

## Current State

- The algebraic simplification pass is enabled and working (#c4ad1c2)
- The folder pass is intentionally excluded in `stablehlo_parser.mm` with a comment explaining why
- All 1474 tests pass without the folder pass

## Reproduction

Enable the folder pass in `runOptimizationPasses()` and run:
```bash
uv run python -c "
import jax, jax.numpy as jnp
a = jnp.eye(2) + jnp.ones((2,2))
jax.grad(lambda a: jnp.linalg.cholesky(a).sum())(a)
"
```
This hangs indefinitely. With `jax.jit` wrapping the grad call, it completes fine.

## Next Steps

- Investigate why the folded constants cause hangs specifically in eager mode
- The IR dumps at `/tmp/stablehlo_pre_opt_*.mlir` and `/tmp/stablehlo_post_opt_*.mlir` (from debug code removed in c4ad1c2) showed the only substantive change was constant+broadcast folding
